### ISSUE TITLE
How to pass extra params to config loader since registration moved to settings.py in 0.18.x

### ISSUE THREAD

**Issue by Mercurrent** [kedro-org/kedro#1527]
## Description

In 0.17.X it was possible to pass params from cli into config loader:
```
class ProjectHooks:
    @hook_impl
    def register_config_loader(
        self,
        conf_paths: Iterable[str],
        env: str,
        extra_params: Dict[str, Any],
    ) -> TemplatedConfigLoader:
        self.extra_params = extra_params
        return TemplatedConfigLoader(conf_paths=conf_paths, globals_pattern="*globals.yml", globals_dict=extra_params)
```
With 0.18.x it doesn't seem to be possible anymore.

## Context
Seems close to:[ issue 1411](https://github.com/kedro-org/kedro/issues/1411)

How has this bug affected you? What were you trying to accomplish?
Needed to be able to pass/override params via CLI

**Comment by noklam** [comment]
@Mercurrent Did you mean `kedro run - - params` is not working for you?

**Comment by Mercurrent** [comment]
Yes, @noklam seems I cannot pass it anymore when I use custom Config Loader. Or I'm missing the way to do this.

Any suggestions for migration of the ProjectHook described?

**Comment by noklam** [comment]
I think ProjectHook is no longer needed. The official way to do it is define TemplateConfigLoader as your loader class in 'settings.py'.

You can find these changes from the RELEASE.md and migration guides.

**Comment by Mercurrent** [comment]
Indeed, it's not needed anymore. How would you migrate the mentioned piece to keep access to params from cli?
We're moving from 0.17.7 onto 0.18.1
TemplateConfigLoader within settings doesn't have access to cli args.

**Comment by noklam** [comment]
@Mercurrent The short answer is you just need to put `TemplatedConfigLoader` in `settings.py` and it should just work.

Before it was moved to `settings.py`, you will need to define it in `ProjectHook`. These parameters are passed from the CLI to the corresponding kedro's class. Under the hood, this is what happened.

https://github.com/kedro-org/kedro/blob/686dba54775ddc7a1ddaef5481e3b82717d33b0f/kedro/framework/session/session.py#L257-L262

**Comment by Mercurrent** [comment]
On 0.17.7 I could use the hook described in the first message above and call a pipeline this way: 
`kedro run --pipeline=data_engineering_pipeline  --params:mode=train` This way mode would override any config params via globals_dict

On 0.18.1
when I use TemplateConfigLoader from settings, there's no way to use CLI params for config params override.

**Comment by noklam** [comment]
@Mercurrent The syntax you provided `kedro run --pipeline=data_engineering_pipeline --params:mode=train` should be still valid for `0.18.1` with `TemplateConfigLoader`.

**Comment by noklam** [comment]
I have created a demo project https://github.com/noklam/kedro_gallery/tree/master/template_config_loader_demo.

`kedro run --params env:RUNTIME,random_state:2022`, you should be able to see these parameters variables get override and printed.

```
**************************************************
{'env': 'RUNTIME',
 'random_state': 2022,
 'target_column': 'species',
 'train_fraction': 0.8}
**************************************************
```

**Comment by lbonini94** [comment]
Hello,
just sharing that I'm having the same issue.
I want to upgrade from 0.17.7 to 0.18.1 but i have a ProjectHooks class with cutom parameters loading method.
```python
class ProjectHooks:
    @hook_impl
    def register_config_loader(
        self, 
        conf_paths: Iterable[str], 
        env: str, 
        extra_params: Dict[str, Any],
    ) -> TemplatedConfigLoader:
    
    extra_parameters = dict()
    ###############
    # custom code #
    ###############

    return TemplatedConfigLoader(
            conf_paths,
            globals_pattern="*globals.yml",
            globals_dict=extra_parameters
        )
    
```

**Comment by AntonyMilneQB** [comment]
Hi @lbonini94, would something like this work for you?
```
class MyTemplatedConfigLoader(TemplatedConfigLoader):
    def __init__(*args, **kwargs):
        extra_parameters = dict()
        ...
        kwargs["globals_dict"] = extra_params
        super().__init__(*args, **kwargs)
```

**Comment by lbonini94** [comment]
Thanks @AntonyMilneQB it works, I just needed to add the self parameter at __init__

**Comment by noklam** [comment]
This is still a very commonly asked question and tricky to find out the solution. I post it in the Discussion since it's easier to pin the solution and hopefully that's clearer and easier to point user to the right place.

There should be a few more but I can't remember all of them, let's add them to the list when the questions come up again.

https://github.com/kedro-org/kedro/issues/1527
